02-parsing/ospf1.robot

```diff
@@ -7,23 +8,56 @@ Resource  cxta.robot
 Suite Setup     suite-setup
 
 *** Test Cases ***
-check ospf neighbours on R1
+
+Check ospf neighbours on R1
     select device "r1"
     run "show ip ospf neighbor"
+    # a simple cxta keyword to check if the output contains a string
     Output contains "FULL"
 
-check ospf neighbours on R2
+Check ospf neighbours on R2
     # using a different way to execute and parse Output
     ${output}=   execute "show ip ospf neighbor" on device "r2"
     # we are using a builtin robot keyword, those typically use
     # tabular arguments where args need to be separated by two or more spaces
     # when using robot text files
     Should Contain    ${output}    FULL
 
+Get the ospf neighbour ID from R1 using parsing keywords
+    select device "r1"
+    run "show ip ospf neighbor"
+    # uses a cxta keyword to find all regex pattern matches on the output from the show command above
+    # the pattern will match IPv4 address, e.g. 192.168.10. All pattern matches will be saved in a list.
+    ${output}=  extract patterns "\\d+\.\\d+\.\\d+\.\\d+"
+    # uses a couple of keywords to log the result of the 'extract patterns' keyword - view these in the log.html
+    log  ${output}
+    log list  ${output}
+    # uses another builtin keyword to set the variable called 'NBR_ID' as the first item in the list.
+    # in this example, the first item is the neighbor ID (lists starts from index 0)
+    ${NBR_ID}=  Set Variable  ${output}[0]
+    # logs the following message along with the neighbor id onto the console
+    log to console  The Neighbor ID is ${NBR_ID}
+
+Get the ospf neighbor ID state from R1
+    # uses a longer regex pattern to retrieve the ospf state
+    # we don't need to use the 'run' keyword again to execute the command as it was
```

Comment:
In the first exercise (01-basic), we highlight that test cases should be indepenent from each other. So can you please re-execute the run command, maybe adding a comment that it it not needed if the whole test suite is executed, but done as best practice..

Reply:
Done, also re-executed 'select device'